crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -361,49 +373,35 @@ impl DWalletMPCManager {
                             mpc_network_key_status,
                             DwalletMPCNetworkKeysStatus::Ready(_)
                         ));
-
-                is_ready
-                    .then(|| is_manager_ready.then_some(session))
-                    .flatten()
-            })
-            .collect::<Vec<&mut DWalletMPCSession>>();
-
-        let mut malicious_parties = vec![];
-        let mut messages = vec![];
-        ready_to_advance
-            .par_iter_mut()
-            .map(|session| {
-                session.round_number += 1;
-                (session.advance(), session.session_info.session_id)
-            })
-            .collect::<Vec<_>>()
-            // Convert back to an iterator for processing.
-            .into_iter()
-            .try_for_each(|(result, session_id)| match result {
-                Ok((message, malicious)) => {
-                    messages.push((message, session_id));
-                    malicious_parties.extend(malicious);
-                    Ok(())
-                }
-                Err(DwalletMPCError::MaliciousParties(malicious)) => {
-                    malicious_parties.extend(malicious);
-                    Ok(())
+                if is_ready && is_manager_ready {
+                    session.round_number = session.round_number + 1;
+                    Some(session.clone())
+                } else {
+                    None
                 }
-                Err(e) => Err(e),
-            })?;
-
-        self.flag_parties_as_malicious(&malicious_parties)?;
+            })
+            .collect();
 
-        // Need to send the messages' one by one, so the consensus adapter won't think they
-        // are a [soft bundle](https://github.com/sui-foundation/sips/pull/19).
-        for (message, session_id) in messages {
-            self.consensus_adapter
-                .submit_to_consensus(&vec![message], &self.epoch_store()?)
-                .await?;
+        for mut session in ready_to_advance.into_iter() {
+            self.ready_to_advance
+                .insert(session.session_info.session_id, session);
         }
         Ok(())
     }
 
+    fn perform_cryptographic_computation(&mut self) {
+        for session in self.ready_to_advance.values().into_iter() {
+            let handle = tokio::runtime::Handle::current();
+            let owned_session_for_different_thread = session.clone();
+            rayon::spawn_fifo(move || {
+                if let Err(err) = owned_session_for_different_thread.advance(handle) {
+                    error!("Failed to advance session with error: {:?}", err);
+                }
+            });
+        }
+        self.ready_to_advance.clear();
```

Comment:
explain the error case, and that this is by design, and there is nothing we can do.